hudi-utilities/src/main/java/org/apache/hudi/utilities/sources/helpers/CloudObjectsDfsSelector.java

```diff
@@ -0,0 +1,285 @@
+     * to use is {@link CloudObjectsDfsSelector}
+     */
+    public static CloudObjectsDfsSelector createSourceSelector(TypedProperties props,
+                                                               Configuration conf) {
+        String sourceSelectorClass = props.getString(CloudObjectsDfsSelector.Config.SOURCE_INPUT_SELECTOR,
+                CloudObjectsDfsSelector.class.getName());
+        try {
+            CloudObjectsDfsSelector selector = (CloudObjectsDfsSelector) ReflectionUtils.loadClass(sourceSelectorClass,
+                    new Class<?>[]{TypedProperties.class, Configuration.class},
+                    props, conf);
+
+            log.info("Using path selector " + selector.getClass().getName());
+            return selector;
+        } catch (Exception e) {
+            throw new HoodieException("Could not load source selector class " + sourceSelectorClass, e);
+        }
+    }
+
+    /**
+     * Get the messages from queue from last checkpoint.
+     *
+     * @param sourceLimit max bytes to read each time
+     * @return the list of files concatenated and their latest modified time
+     */
+    public Pair<Option<String>, String> getNextFilePathsFromQueue(Option<String> lastCheckpointStr, long sourceLimit) {
+        System.out.println("Reading messages....");
+
+//        String queueUrl = sqs.getQueueUrl(this.queueUrl).getQueueUrl();
+
+        try {
+            System.out.println("lastCheckpointStr" + lastCheckpointStr);
+
+            long lastCheckpointTime = lastCheckpointStr.map(Long::parseLong).orElse(Long.MIN_VALUE);
+
+            System.out.println(lastCheckpointTime);
+
+            List<JSONObject> eligibleFiles = listFilesAfterCheckpoint(this.sqs, this.queueUrl, lastCheckpointTime);
+            System.out.println("eligibleFiles size: " + eligibleFiles.size());
+
+            // sort all files by event time.
+            eligibleFiles.sort(Comparator.comparingLong(record -> record.getLong("eventTimeLong")));
+
+            List<String> filteredFiles = new ArrayList<>();
+            long currentBytes = 0;
+            long newCheckpointTime = lastCheckpointTime;
+
+            for (JSONObject record : eligibleFiles) {
+
+                long eventTime = record.getLong("eventTimeLong");
+                JSONObject s3Object = record.getJSONObject("s3").getJSONObject("object");
+                String bucket = URLDecoder.decode(record.getJSONObject("s3").getJSONObject("bucket").getString("name"), "UTF-8");
+                String key = URLDecoder.decode(s3Object.getString("key"), "UTF-8");
+                String filePath = "s3://" + bucket + "/" + key;
+
+                String fileName = StringUtils.substringAfterLast(key, "/");
+//                System.out.println(eventTime);
+
+                // toDO - move this check to listFilesAfterCheckpoint function
+                if (IGNORE_FILEPREFIX_LIST.stream().noneMatch(fileName::startsWith)) {
```

Comment:
yeah let's move this to `listFilesAfterCheckpoint ` method as you have mentioned.